docs/gen-versions.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import os
+import subprocess
+import sys
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        prog="gen-versions.py",
+        description="Generate Sphinx version switcher JSON file from git tags.",
+    )
+
+    parser.add_argument(
+        "-o",
+        "--out-file",
+        help="path to output file, including filename, for generated versions JSON file",
+        metavar="path",
+        default=None,
+    )
+
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+
+    completed = subprocess.run(["./gen-versions.sh"], capture_output=True)
+    versions = completed.stdout.splitlines()
+
+    output = []
+
+    # Special case for latest.
+    latest = versions.pop(0)
+    latest = latest.decode()
```

Comment:
i don't really know why it's encoded, and i don't know what latest looks like or why it needs to be changed.